### DISCUSSION TITLE
How to read all the text in a page excluding page header/footer.

### DISCUSSION THREAD

**Discussion by Julio-German-Gutierrez** [UglyToad/PdfPig#404] [Q&A] [answered]
Hey I just discovered your library and it is great!
I am working on a small project to perform simple resume of pdf-text documents. So far I have been able to deal with page header/footer, using regular expressions. As you can imagine this is far from good as it has to be personalized with every new document. I am currently using the "text" property of a page `myPage.text` to access text content. The problem is that I am also getting header & footer.

Is there a way to get only text-content without header/footer?

Thanks a lot!

**Comment by BobLd** [comment, marked as answer]
Hi @Julio-German-Gutierrez, a first step would be to identify headers and footers (as you try to do with regex).
Have a look in the Wiki here: [Decoration Text Block Classifier](https://github.com/UglyToad/PdfPig/wiki/Document-Layout-Analysis#decoration-text-block-classifier) for another method to find headers and footers. This is far from perfect but this is a start...

You'll then have to remove them from your text (I strongly advise you have a look at the whole Document Layout Analysis wiki page for a better understanding)

**Reply by Julio-German-Gutierrez** [reply]
Thanks a lot!

**Reply by BobLd** [reply]
Have a look there [page segmenters](https://github.com/UglyToad/PdfPig/wiki/Document-Layout-Analysis#page-segmenters)

**Reply by Julio-German-Gutierrez** [reply]
It works great thanks!
